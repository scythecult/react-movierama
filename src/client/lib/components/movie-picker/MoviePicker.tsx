import type { ChangeEvent } from 'react';
import { STORED_MOVIE_KEY } from '../../constants/common';
import styles from './styles.module.css';

const movies = [
  { title: 'Avengers: Endgame', price: '10' },
  { title: 'Joker', price: '12' },
  { title: 'Toy Story 4', price: '8' },
  { title: 'The Lion King', price: '9' },
  { title: 'Matrix', price: '1' },
];

export const MoviePicker = () => {
  const selectedMovie = localStorage.getItem(STORED_MOVIE_KEY) || '';

  const handleChange = (evt: ChangeEvent<HTMLSelectElement>) => {
    const movieTitle = evt.target.value;
    const selectedMovie = movies.find((movie) => movie.title === movieTitle);

    localStorage.setItem(STORED_MOVIE_KEY, JSON.stringify(selectedMovie));
  };

  return (
    <div className={styles.moviePicker}>
      <label htmlFor="movie">Pick a&nbsp;movie:</label>
      <select id="movie" value={selectedMovie} onChange={handleChange}>
        {movies.map((movie) => {
          const { title, price } = movie;
          return (
            <option key={title} value={title}>
              {title} (${price})
            </option>
          );
        })}
      </select>
    </div>
  );
};
