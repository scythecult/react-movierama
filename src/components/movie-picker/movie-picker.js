import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { STORED_MOVIE_KEY } from "../../consts/consts";
import { setPrice } from "../../features/movies/movies-slice";
import classes from "./movie-picker.module.css";

const MoviePicker = () => {
  const { movies, selectedMovie } = useSelector((state) => state.moviePicker);
  const dispatch = useDispatch();

  const handleChange = (evt) => {
    const movieTitle = evt.target.value;

    dispatch(setPrice({ selectedMovie: movieTitle }));
    localStorage.setItem(STORED_MOVIE_KEY, movieTitle);
  };

  return (
    <div className={classes["movie-picker"]}>
      <label htmlFor="movie">Pick a&nbsp;movie:</label>
      <select id="movie" value={selectedMovie} onChange={handleChange}>
        {movies.map((movie) => {
          const { title, price } = movie;
          return (
            <option key={title} value={title}>
              {title} (${price})
            </option>
          );
        })}
      </select>
    </div>
  );
};

export { MoviePicker };
