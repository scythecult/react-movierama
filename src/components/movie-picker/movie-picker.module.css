.movie-picker {
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 20px 0;
}

.movie-picker select {
  background-color: #fff;
  border: 0;
  border-radius: 5px;
  font-size: 14px;
  margin-left: 10px;
  padding: 5px 15px 5px 15px;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
}
