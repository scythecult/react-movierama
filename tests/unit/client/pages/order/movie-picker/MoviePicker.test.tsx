import { render } from '@testing-library/react';
import { MoviePicker } from '@/client/pages/order/movie-picker/MoviePicker';

describe('MoviePicker', () => {
  test('should correspond default layout', () => {
    const result = render(<MoviePicker />);

    expect(result.container).toMatchSnapshot();
  });
});
