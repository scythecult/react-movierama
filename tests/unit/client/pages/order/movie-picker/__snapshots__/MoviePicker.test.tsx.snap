// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MoviePicker > should correspond default layout 1`] = `
<div>
  <div
    class="_moviePicker_b8b8f9"
  >
    <label
      for="movie"
    >
      Pick aÂ movie:
    </label>
    <select
      id="movie"
    >
      <option
        value="Avengers: Endgame"
      >
        Avengers: Endgame
         ($
        10
        )
      </option>
      <option
        value="Joker"
      >
        Joker
         ($
        12
        )
      </option>
      <option
        value="Toy Story 4"
      >
        Toy Story 4
         ($
        8
        )
      </option>
      <option
        value="The Lion King"
      >
        The Lion King
         ($
        9
        )
      </option>
      <option
        value="Matrix"
      >
        Matrix
         ($
        1
        )
      </option>
    </select>
  </div>
</div>
`;
